src/User/Model/User.php

```diff
@@ -248,7 +250,9 @@ public function rules()
             // two factor auth rules
             'twoFactorSecretTrim' => ['auth_tf_key', 'trim'],
             'twoFactorSecretLength' => ['auth_tf_key', 'string', 'max' => 16],
-            'twoFactorEnabledNumber' => ['auth_tf_enabled', 'boolean']
+            'twoFactorEnabledNumber' => ['auth_tf_enabled', 'boolean'],
+            'twoFactorTypeLength' => ['auth_tf_type', 'string', 'max' => 20],
+            'twoFactorTypeLength' => ['auth_tf_mobile_phone', 'string', 'max' => 20],
```

Comment:
The rule of `auth_tf_mobile_phone` overrides the one of `auth_tf_type` since it has the same key, please change it to `twoFactorMobilePhoneLength`